tests/test_utils.py

```diff
@@ -425,6 +425,12 @@ def test_validate_argument():
         assert validate_argument(loc, argname, ('float_array', str), argmin=0, argmax=4,
                                  eqmin=True, eqmax=True) is None
 
+    # Test shape
+    for argname in ('int', 'float', 'str') :
+        assert validate_argument(loc, argname, (str, 'float_array'), shape=()) is None
+    for argname in ('int_array', 'float_array') :
+        assert validate_argument(loc, argname, 'float_array', shape=(2,)) is None
+
     assert_raises(TypeError, validate_argument, loc, 'str', ('float_array', str), argmin=0)
 
```

Comment:
Hi @rancesol , what you also want here is to make sure the `validate_argument` raises an error when the shape is wrong. Try using a `assert_raises` test